build-tools/packages/build-cli/test/commands/generate/buildVersion.test.ts

```diff
@@ -345,3 +345,229 @@ describe("generate:buildVersion", () => {
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "prerelease",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "untrimmed",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("tagName: client, release: prerelease, types: untrimmed", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.4.0-88879");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "release",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "untrimmed",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("tagName: client, release: release, types: untrimmed", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.4.0");
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "release",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "public",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("tagName: client, release: release, types: public", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.4.0");
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "release",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "alpha",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("tagName: client, release: release, types: alpha", (ctx) => {
+			expect(ctx.stdout).to.contain(
+				`ERROR: ERROR: This release type is not supported. Alpha/beta ***prereleases*** are allowed.`,
+			);
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "false",
+		VERSION_RELEASE: "release",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "beta",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("tagName: client, release: release, types: beta", (ctx) => {
+			expect(ctx.stdout).to.contain(
+				`ERROR: ERROR: This release type is not supported. Alpha/beta ***prereleases*** are allowed.`,
+			);
+		});
+
+	test.env({
+		VERSION_BUILDNUMBER: "88879",
+		VERSION_TAGNAME: "client",
+		TEST_BUILD: "true",
+		VERSION_RELEASE: "prerelease",
+		VERSION_PATCH: "False",
+		VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		PACKAGE_TYPES_FIELD: "alpha",
+	})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "0.4.0"])
+		.it("tagName: client, release: prerelease, test-build: true, types: alpha", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.0.0-88879-alpha-types-test");
```

Comment:
sorry about that. The versioning for test feed should be the same. fixing this

Reply:
> Where is `codeVersion` used in the pipeline: [`main`/tools/pipelines/templates/include-set-package-version.yml#L105](https://github.com/microsoft/FluidFramework/blob/main/tools/pipelines/templates/include-set-package-version.yml?rgh-link-date=2023-10-17T19%3A31%3A38Z#L105)?

I'm not sure how that is used. @tyler-cai-microsoft might know.

Reply:
`codeVersion` is used in our e2e tests when we run `backCompat` tests. And for our test branches especially since their versioning scheming is a little weird as they are expected to be `0.0.0-xyz-test`, which would fail semver comparisons.

Reply:
so `CODEVERSION` causes these two test to fail: https://dev.azure.com/fluidframework/internal/_build/results?buildId=199804&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=ca111d9b-56f7-5b3d-b8f6-661164061440